extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryFrameProcessor.java

```diff
@@ -484,7 +479,8 @@ private void convertRowFrameToRowsAndColumns(Frame frame)
 
   /**
    * Compare two rows based on the columns in partitionColumnNames.
-   * If the partitionColumnNames is empty or null, compare entire row.
+   * If the partitionColumnNames is null, compare entire row.
```

Comment:
why add that extra meaning to null? its not really realistic to think that all columns will be partition columns...

I don't think its used&tested - instead of allowing null make sure that the `partitionColumnNames` field is never null after the constructor was called - so that there is no chance to get an NPE in L187 either

Reply:
Makes sense, have made the change.